backend/packages/Upgrade/src/api/services/AnalyticsService.ts

```diff
@@ -188,7 +193,51 @@ export class AnalyticsService {
         if (experiment.assignmentUnit === ASSIGNMENT_UNIT.WITHIN_SUBJECTS) {
           csvExportData = await this.analyticsRepository.getCSVDataForWithInSubExport(experimentId, skip, take);
         } else {
-          csvExportData = await this.analyticsRepository.getCSVDataForSimpleExport(experimentId, skip, take);
+          // Get the experiment details
+          const experimentRepository = getCustomRepository(ExperimentRepository, 'export');
+          const experimentQuery = experimentRepository
+            .createQueryBuilder('experiment')
+            .select([
```

Comment:
This same query will now potentially be run multiple times to get the same data, since it is inside the 'do...while' loop. However, we're considering getting rid of that loop altogether anyway. But if we keep the query, it must be moved outside the loop.

Reply:
Yeah, it was a mistake. I had made the change for moving it outside the do while loop and was waiting for other changes and have now pushed that in my last commit. But, yeah you told we want to remove the loop for skip take.